tICA/scripts/feature_helpers/brain_masks.m

```diff
@@ -0,0 +1,151 @@
+end
+VolumeTemplate = [GroupAverageName '_CIFTIVolumeTemplate_' OutputfMRIName '.' Resolution '.dscalar.nii'];
+
+if isempty(Dilate)
+  Dilate = '5.0';
+end
+
+resolution=str2double(Resolution);
+erode_mm=cell(1,5);
+for i=1:5
+    erode_mm{i}=num2str(resolution*i);
+end
+% if strcmp(Resolution, '2')
+%     erode_mm = {'2', '4', '6', '8', '10'};
+% elseif strcmp(Resolution, '1.60')
+%     erode_mm = {'1.60', '3.20', '4.80', '6.40', '8'};
+% else
+%     error('Please specify resolution as 2 or 1.60')
+% end
+
+EssentialFileNames={[MaskSavePath '/' GroupAverageName '_AverageWM_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageGM_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageCSF_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageSubcortical_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageRightCerebellar_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageLeftCerebellar_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageLeftGM_NoCerebellar_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_AverageRightGM_NoCerebellar_Mask_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_brainboundary_' erode_mm{1} 'mm_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_brainboundary_' erode_mm{2} 'mm_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_brainboundary_' erode_mm{3} 'mm_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_brainboundary_' erode_mm{4} 'mm_' OutputfMRIName '.' Resolution '.dscalar.nii'], ...
+                    [MaskSavePath '/' GroupAverageName '_brainboundary_' erode_mm{5} 'mm_' OutputfMRIName '.' Resolution '.dscalar.nii']};
+
+% check if mask files are already there
+ToRun=0;
+for i=1:length(EssentialFileNames)
+    if ~isfile(EssentialFileNames{i})
+        ToRun=1;
+    end
+end
+
+% only generate mask files when there's missing file
+if ToRun==1
+    MNIPath=[StudyFolder '/' GroupAverageName '/MNINonLinear'];
+
+    if ~exist(MaskSavePath, 'dir')
+        mkdir(MaskSavePath);
+    end
+
+    % specific brain region masks
+    unix(['wb_command -volume-label-import ' MNIPath '/' GroupAverageName '_Averagewmparc.nii.gz ' ConfigFilePath '/FreeSurferAllLut.txt ' MaskSavePath '/' GroupAverageName '_Averagewmparc_for_mask.nii.gz -drop-unused-labels']);
```

Comment:
This looks like a shell script written as a matlab function (and therefore without error handling).  Maybe consider rewriting it in bash if it isn't going to actually load any files.

Reply:
Yes, I agree that it should be changed into a shell script.